.github/workflows/torchserve-nightly-build.yml

```diff
@@ -6,28 +6,38 @@ on:
     - cron:  '15 11 * * *'
 jobs:
   nightly:
-    runs-on: ubuntu-18.04
+    # creates workflows for the 3 OS: ubuntu, macOS & windows
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macOS-latest, windows-latest]
     steps:
-      - name: Setup Python 3.8
-        uses: actions/setup-python@v2
-        with:
-          python-version: 3.8
-          architecture: x64
+      - name: Setup Conda
+        uses: s-weigand/setup-conda@v1
+      - run: conda --version
+      - run: conda install -y conda-build
       - name: Checkout TorchServe
         uses: actions/checkout@v2
       - name: Install dependencies
         run: |
-          set -eux
           pip install twine
-          sudo apt-get update
-          python ts_scripts/install_dependencies.py --cuda=cu102 --environment=dev
+          python ts_scripts/install_dependencies.py --environment=dev
           pip install -e .
       - name: Setup Java 11
         uses: actions/setup-java@v2
         with:
           distribution: 'zulu'
           java-version: '11'
-      - name: Push nightly
+      - name: conda config for auto-upload
+        run: conda config --set anaconda_upload yes
+      - name: Build PyPI & conda nightly binaries and push conda binaries
+        env:
+          ANACONDA_API_TOKEN: ${{ secrets.CONDA_NIGHTLY_TOKEN }}
+        run: python binaries/build.py --nightly
+      - name: Push PyPI nightly binaries
         env:
           PYPI_TOKEN: ${{ secrets.PYPI_TOKEN }}
-        run: ts_scripts/push_nightly.sh torchserve-nightly
+        # skip pushing PyPI nightly binaries for macOS & windows as the binaries have the same name
+        if: matrix.os == 'ubuntu-latest'
+        run: python binaries/upload.py --upload-pypi-packages --pypi-token "$PYPI_TOKEN"
```

Comment:
Let's stick to one convention with passing tokens either as arguments to function or as environment variables. IMO environment variables are safer since you don't risk accidentally printing the token in the github action console at which point the credentials need to be recycled which is a big hassle

Reply:
Changed it to use env variable